### PR TITLE
Remove `reduction_identity<bool>::{sum,prod,max,min}`

### LINKED ISSUE #4707: Nightly build error gcc-10 -Werror: TestReducers.hpp:379:22: TestReducers.hpp error: ‘*’ in boolean context
gcc-10 builds are failing with -Werror with output:

```
/home/jenkins/blake-new/workspace/Kokkos_develop_blake_gcc_openmp_pthread_serial_1020-dbg/kokkos/core/unit_test/TestReducers.hpp:379:22: error: ‘*’ in boolean context, suggest ‘&&’ instead [-Werror=int-in-bool-context]
  379 |       reference_prod *= h_values(i);
```

**Reproducer** (blake):
```
module load cmake/3.19.3 gcc/10.2.0

$KOKKOS_PATH/generate_makefile.bash --with-devices=OpenMP --arch=SKX --compiler=g++ --cxxflags="-g -Wall -Wunused-parameter -Wshadow -pedantic -Werror -Wsign-compare -Wtype-limits -Wignored-qualifiers -Wempty-body -Wclobbered -Wuninitialized " --cxxstandard="14" --kokkos-path=$KOKKOS_PATH --no-examples --with-options=enable_large_mem_tests --debug
```

This began after adding boolean reducer tests

### PR DESCRIPTION
**Pull request by ajpowelsnl**
Fix #4707

Changes here remove all `reduction_identity` specializations for `bool`, except for `lor` and `land.`  Builds on Blake (Sandia Intel Skylake machine) now succeed with `cmake-3.19.3`, `binutils/2.30.0` and `gcc-10.2.0` (with the ` -Werror` flag).

### CODE REVIEW

File: core/unit_test/TestReducers.hpp [hunk 1]
```diff
@@ -1036,10 +1028,6 @@ struct TestReducers {
   static void execute_bool() {
     test_LAnd(10001);
     test_LOr(35);
-    test_sum(10001);
-    test_prod(35);
-    test_min(10003);
-    test_max(10007);
```
Comment (masterleinad) [inline core/unit_test/TestReducers.hpp:1042]: This should be the only change necessary in this file. We shouldn't delete the tests for other scalar types above.

File: core/src/Kokkos_NumericTraits.hpp [hunk 2]
```diff
@@ -364,16 +364,9 @@ struct reduction_identity<signed char> {
   }
 };
 
+// 4707 fix; effectively reverting much of PR #4352
```
Comment (masterleinad) [inline core/src/Kokkos_NumericTraits.hpp:367]: I don't think it adds any value and would just remove.
```suggestion
```

**Review by masterleinad — CHANGES_REQUESTED** [review]
You will also need to run the indentation script:
```diff
+ ./scripts/docker/check_format_cpp.sh
diff --git a/core/src/Kokkos_NumericTraits.hpp b/core/src/Kokkos_NumericTraits.hpp
index a9bd732ef..fa0c85ec7 100644
--- a/core/src/Kokkos_NumericTraits.hpp
+++ b/core/src/Kokkos_NumericTraits.hpp
@@ -364,7 +364,7 @@ struct reduction_identity<signed char> {
   }
 };
 
-// 4707 fix; effectively reverting much of PR #4352 
+// 4707 fix; effectively reverting much of PR #4352
 template <>
 struct reduction_identity<bool> {
   KOKKOS_FORCEINLINE_FUNCTION constexpr static bool lor() {
diff --git a/core/unit_test/TestReducers.hpp b/core/unit_test/TestReducers.hpp
index 3a9061d2c..1e9846e43 100644
--- a/core/unit_test/TestReducers.hpp
+++ b/core/unit_test/TestReducers.hpp
@@ -398,9 +398,9 @@ struct TestReducers {
       ASSERT_EQ(prod_scalar, reference_prod);
 
       prod_scalar = init;
-      /*Kokkos::parallel_reduce(Kokkos::RangePolicy<ExecSpace, ReducerTag>(0, N),
-                              f_tag, reducer_scalar);*/
-      //ASSERT_EQ(prod_scalar, reference_prod);
+      /*Kokkos::parallel_reduce(Kokkos::RangePolicy<ExecSpace, ReducerTag>(0,
+         N), f_tag, reducer_scalar);*/
+      // ASSERT_EQ(prod_scalar, reference_prod);
 
       Scalar prod_scalar_view = reducer_scalar.reference();
       ASSERT_EQ(prod_scalar_view, reference_prod);
script returned exit code 1
```

File: core/unit_test/TestReducers.hpp (hunk 1 above)
Comment (crtrott) [reply core/unit_test/TestReducers.hpp:1042]: What Daniel said.

**Review by crtrott — CHANGES_REQUESTED** [review]

File: core/src/Kokkos_NumericTraits.hpp (hunk 2 above)
Comment (ajpowelsnl) [reply core/src/Kokkos_NumericTraits.hpp:367]: Oh, right.  That was just a "note to self" that I forgot to remove.

**Review by crtrott — APPROVED** [review]

**Review by dalg24 — APPROVED** [review]

**Pull request merged**